.github/workflows/glog.yml

```diff
@@ -0,0 +1,102 @@
+name: glog
+
+on:
+  push:
+    branches:
+      - '*'
+    path:
+      - 'instrumentation/glog/**'
+      - '.github/workflows/glog.yml'
+  pull_request:
+    branches: [main]
+    paths:
+      - 'instrumentation/glog/**'
+      - '.github/workflows/glog.yml'
+
+jobs:
+  cmake_linux:
+    name: CMake Linux
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout googletest
+        uses: actions/checkout@v3
+        with:
+          repository: "google/googletest"
+          ref: "release-1.12.1"
+          path: "googletest"
+          submodules: "recursive"
+      - name: checkout glog
+        uses: actions/checkout@v3
+        with:
+          repository: "google/glog"
+          # TODO: upgrade to v0.7.0 and fix deprecation warnings
```

Comment:
One problem with glog is that it appears to be unversioned and the only way to figure out when something is changed in the API has been by basing it on a specific define that appears when the breakage happens, like [this](https://github.com/open-telemetry/opentelemetry-cpp-contrib/blob/main/instrumentation/glog/include/opentelemetry/instrumentation/glog/sink.h#L15C1-L20C7)

Possibily, something similar would need to be done for the deprecation notice around `.timestamp()` too

Reply:
I used 0.6.0 because of the timestamp deprecation warnings in 0.7.0

This PR is just to add CI, trying to limit the changes to a minimum.

After this, I think we can upgrade to 0.7.0, change the timestamp code to resolve warnings, and require 0.7.0 as a minimum then, also resolving the ifdef around GLOG_VERSION_HAS_LOGMESSAGETIME.

Looking at the glog source, the deprecated methods are removed already from version 0.8.0 to come, so this must be fixed eventually.

Reply:
Resolved by upgrading to glog 0.7.0, now required, and fixing deprecation warnings.